nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,849 @@
+
+    screening_percentile : int, float, optional, in the closed interval [0, 100]
+        Perform an univariate feature selection based on the Anova F-value for
+        the input data. A float according to a percentile of the highest
+        scores. Default: 20.
+
+    scoring : str, callable or None, optional. Default: 'roc_auc'
+        The scoring strategy to use. See the scikit-learn documentation
+        If callable, takes as arguments the fitted estimator, the
+        test data (X_test) and the test target (y_test) if y is
+        not None.
+        e.g. scorer(estimator, X_test, y_test)
+
+        For classification: 'accuracy', 'f1', 'precision', 'recall' or 'roc_auc'
+            Default 'roc_auc'.
+
+    smoothing_fwhm : float, optional. Default: None
+        If smoothing_fwhm is not None, it gives the size in millimeters of the
+        spatial smoothing to apply to the signal.
+
+    standardize : bool, optional. Default: True
+        If standardize is True, the time-series are centered and normed:
+        their variance is put to 1 in the time dimension.
+
+    target_affine : 3x3 or 4x4 matrix, optional. Default: None
+        This parameter is passed to image.resample_img. Please see the
+        related documentation for details.
+
+    target_shape : 3-tuple of int, optional. Default: None
+        This parameter is passed to image.resample_img. Please see the
+        related documentation for details.
+
+    low_pass : None or float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details
+
+    high_pass : None or float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details
+
+    t_r : float, optional. Default: None
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details.
+
+    mask_strategy : {'background' or 'epi'}, optional. Default: 'background'
+        The strategy used to compute the mask: use 'background' if your
+        images present a clear homogeneous background, and 'epi' if they
+        are raw EPI images. Depending on this value, the mask will be
+        computed from masking.compute_background_mask or
+        masking.compute_epi_mask.
+
+    memory : instance of joblib.Memory or str
+        Used to cache the masking process.
+        By default, no caching is done. If a str is given, it is the
+        path to the caching directory.
+
+    memory_level : int, optional. Default: 0
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    n_jobs : int, optional. Default: 1.
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs'.
+
+    verbose : int, optional. Default: 0.
+        Verbosity level.
+
+    See Also
+    ------------
+    nilearn.decoding.Decoder - classification strategies for Neuroimaging.
```

Comment:
I don't understand why the decoder is referencing itself in the see also.

We should probably have the DecoderRegressor here, and also the SpaceNetClassifier, no?